File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
from collections import defaultdict

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = defaultdict(int)

        for num in nums:
            freq[num] += 1

        sorted_keys = sorted(freq, key=freq.get, reverse=True)[:k]
        return sorted_keys

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findDuplicate(nums: number[]): number {
        let set = new Set<number>

        for (let num of nums) {
            if (set.has(num)) {
                return num
            }
            set.add(num)
        }
        return -1
    }
}
